### PR TITLE
Improved rules for expansion (issue #20, revisited).

### PR DESCRIPTION
**Pull request by jeff-cohere**
I thought about this some more, and decided to make the 3-item list expansion rules a bit more specific, just to make sure we're not preventing anyone from trying to articulate an actual 3-item sequence.

@pbosler , can you take a look and see if this appeals to you?

### CODE REVIEW

File: docs/input.md [hunk 1]
```diff
@@ -166,22 +166,21 @@ How does Skywalker know this isn't a list containing the temperatures 230.15 K,
...
+A list with values $\[v_1, v_2, v_3\]$ that satisfies the following properties is
+expanded into a uniformly spaced set of values between $v_1$ and $v_2$ with
+spacing $v_3$:
 
-* `v1 < v2` and `v3 < v2`, **OR**
-* `v1 < v2 < 0` and `|v3| < |v2 - v1|`.
+* $v_1 < v_2$
+* $0 < v_3 < v_2$ **or** $v_2 < 0$ and $0 < v_3 < (v_2 - v_1)/2$
 
 All other lists are interpreted as lists containing 3 values.
```
Comment (pbosler) [inline docs/input.md:177]: A few examples here would be helpful:

```
list [0, 1, 0.1] expands to [0.  0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1. ]
list [-2, 0, 0.2] does not expand
list [-2, 1, 0.25] expands to [-2.   -1.75 -1.5  -1.25 -1.   -0.75 -0.5  -0.25  0.    0.25  0.5   0.75
  1.  ]
list [-10, -1, 1] expands to [-10.  -9.  -8.  -7.  -6.  -5.  -4.  -3.  -2.  -1.]
list [-11, 1, 1] does not expand
list [-11, 1, 2] does not expand
list [-7, 2, 1] expands to [-7. -6. -5. -4. -3. -2. -1.  0.  1.  2.]
```

btw: the above is the output from this python script:

```
import numpy as np

def cond1(l):
  return (0 < l[2] and l[2] < l[1])

def cond2(l):
  return (l[1]<0 and 0 < l[2] and l[2] < (l[1] - l[0]))

def list_expand(l):
  result = np.array(l)
  if len(l) == 3:
    if l[0] < l[1]:
      if cond1(l) or cond2(l):
        result = np.linspace(l[0], l[1], int(np.ceil((l[1]-l[0])/l[2]))+1)
  return result

if __name__ == '__main__':
  lists = [ [0, 1, 0.1],
            [-2, 0, 0.2],
            [-2, 1, 0.25],
            [-10, -1, 1],
            [-11, 1, 2],
            [-11, 1, 1],
            [-11, 1, 2],
            [-7, 2, 1],
  ]
  for l in lists:
    el = list_expand(l)
    if len(el) > 3:
      print("list {} expands to {}".format(l, el))
    else:
      print("list {} does not expand".format(l))
```

File: docs/input.md [hunk 2]
```diff
@@ -166,22 +166,21 @@ How does Skywalker know this isn't a list containing the temperatures 230.15 K,
...
+These rules handle most cases of interest, but there remain some sequences that
+suggest expanѕion but that Skywalker does not expand. For example, the list
 `[-11, 1, 2]` could be interpreted as the uniformly-spaced set of values
-`[-11, -9, -7, -5, -3, -1, 1]`, but it is also an ascending set of 3 values,
-so Skywalker doesn't expand it. Because this expanѕion feature is intended as a
-convenience, you can always explicitly perform the expanѕion yourself when you
-create an input file.
+`[-11, -9, -7, -5, -3, -1, 1]`, but since $2 > 1$ and $1 > 0$, Skywalker doesn't
+expand it. Because this expanѕion feature is intended as a convenience, you can
+always explicitly perform the expanѕion yourself when you create an input file.
```
Comment (pbosler) [inline docs/input.md]: I suggest adding emphasis, either bold font or a markup box

**Review by pbosler — APPROVED** [review]

File: docs/input.md (hunk 2 above)
Comment (jeff-cohere) [reply docs/input.md]: Do you mean you want to add emphasis markup for the examples? I'll see what I can come up with.

Comment (pbosler) [reply docs/input.md]: >you can
always explicitly perform the expanѕion yourself when you create an input file.

I just meant that this sentence should be emphasized with bold font or some other means

**Pull request merged**